internal/cmd/internal/olmv1/catalog/alpha_inspect.go

```diff
@@ -0,0 +1,72 @@
+package catalog
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/operator-framework/kubectl-operator/internal/pkg/catalogd/fetcher"
+	"github.com/operator-framework/kubectl-operator/internal/pkg/catalogd/streamer"
+	experimentalaction "github.com/operator-framework/kubectl-operator/internal/pkg/experimental/action"
+	"github.com/operator-framework/kubectl-operator/pkg/action"
+	"sigs.k8s.io/yaml"
+
+	"github.com/spf13/cobra"
+)
+
+func NewInspectCommand(cfg *action.Configuration) *cobra.Command {
+	i, err := experimentalaction.NewCatalogInspect(cfg,
+		func(c *action.Configuration) experimentalaction.CatalogFetcher {
+			return fetcher.New(c.Client)
+		},
+		func(c *action.Configuration) experimentalaction.CatalogContentStreamer {
+			return streamer.New(c.Clientset.CoreV1())
+		},
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+	i.Logf = log.Printf
+
+	cmd := &cobra.Command{
+		Use:   "inspect [name]",
+		Short: "inspect catalog objects",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
Generally speaking, it's best to try to factor CLI commands into a couple of phases - first, gather user input into some `struct{}`, then do any transformations necessary (string parsing, etc), validate the resulting data, and operate on it. This structure helps to make the commands composable but also makes it fairly trivial to use the "meat" of the commands as a library and test them programmatically. See the `kubectl create clusterrole` code as an example of this. 

https://github.com/kubernetes/kubernetes/blob/c4feb19195009e12fa7c49ca1fe91b8898360b0a/staging/src/k8s.io/kubectl/pkg/cmd/create/create_clusterrole.go

Reply:
I don't disagree. under the hood that is _mostly_ what this is doing. For context underneath the hood it:
- Fetches all catalogs
- Filters all catalogs based on some criteria
- For each catalog (post filtering) stream the contents of the catalog
- For processing, during the streaming of the contents, the contents are appended to the list of catalog objects to return if they meet all the filter criteria based on the provided user input via flags or positional arguments.

As much as possible I was trying to follow the same approach that existed prior to me adding these commands (i.e: https://github.com/operator-framework/kubectl-operator/blob/a6eb73a477ceb2295eb9422ccc7b1e7c23db4756/internal/cmd/operator_install.go#L16-L36)

I _personally_ think there is some restructuring of this repo that could be done to make it a bit more intuitive on how to add subcommands and extend the functionality of this plugin but that is out of scope for this PR

Reply:
I'm only speaking to the implementations that are new here - for instance, there's no way to test the post-processing after `Run()` here nor are any invariants validated about `experimentalaction.CatalogInspectOptions` - if you can factor this to collapse the `*cobra.Command` run func to nothing but passing opts elsewhere, which you are super close to, I think you'd find it easier to validate your commands programmatically.